yumpkg: report stdout when stderr is redirected to stdout

marmarek opened the pull request:
### What does this PR do?

When 'redirect_stderr=True' is set, stderr is empty and both
stdout+stderr are logged into stdout. Include the right one in the
output.

### What issues does this PR fix or reference?
Fixes: 57862

### Previous Behavior
Empty `errors` field on errors.

### New Behavior
`errors` field populated.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltstack.com/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

marmarek committed:
pkg: report stdout when stderr is redirected to stdout

When 'redirect_stderr=True' is set, stderr is empty and both
stdout+stderr are logged into stdout. Include the right one in the
output.

Fixes b40e21284b7b0e0accc41abfd1680c5551e2bc7b
Fixes #57862

cmcmarrow commented:
@marmarek I know it’s a little redundant but can we get a test to verify that does not turn back into a regression again? Also thanks for the PR! The failing test are not related to you at this time. It’s a problem on are end.

cmcmarrow reviewed (changes requested):
I think it would be good to include `stdout` and `stderr`. I think someone changed it to `stdout` because they were not getting info need from `stderr`

marmarek commented:
> I think it would be good to include `stdout` and `stderr`

In this specific case it doesn't make sense, as stderr is redirected to stdout. Maybe two other cases should also have `redirect_stderr=True` instead of logging stdout and stderr separately?

terminalmage commented:
@marmarek Yeah, I think I mistakenly "corrected" the line you're fixing in this PR, I should have left it alone. Can you revert the other two from b40e2128 but also add `redirect_stderr=True` to the `_call_yum()` function call? That seems like the best fix here.

terminalmage commented:
@marmarek Just checking back in since I haven't heard back from you. Can you make those changes I mentioned in my last comment?

marmarek committed:
Add test for yumpkg.install error reporting

Make yumpkg module call 'cat' but with yum's options. This will fails
because 'cat' does not understand '-y' option. Assert for proper error
reporting.
Run the test on Linux only (to have specific error message) - yumpkg
doesn't makes sense on non-Linux anyway.

Regression test for #57862

marmarek commented:
I've finally found some time to finish this, sorry for the delay.

terminalmage commented:
Thanks! This looks good, just need a few fixes identified by pre-commit.

marmarek commented:
pre-commit seems to look at the whole file, not only changed parts. Should I include reformatting and various other updates like removing `six` in the commit? Sounds unrelated to this PR...

terminalmage commented:
I know that the core team will not merge this without the tests failing. It seems that the failures are related to some automated code reformatting to ditch usage of six and associated code conventions that ensure Python 2.7 compatibility, now that Python 2 support has been dropped.

Perhaps one of the core team can apply the pre-commit hooks and push to this PR branch.

marmarek committed:
yumpkg: drop six usage, reformat file

Apply changes suggested by pre-commit. This is mostly about dropping six
and python2.

Ch3LL committed:
run pre-commit

Ch3LL commented:
I went ahead and ran pre-commit and fixed some merge conflicts on rebase.

Ch3LL commented:
ping @terminalmage would you mind re-reviewing since I have technically pushed code to this now would be nice to get one more review.

terminalmage reviewed (dismissed):
Looks great :+1:

Ch3LL committed:
Add changelog
